src/lib/containers/QuerySortSelector.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react'
+import { SortConfiguration } from './CardContainerLogic'
+import { useInfiniteQueryContext } from './QueryContext'
+import { SortDirection, SortItem } from '../utils/synapseTypes'
+import { unCamelCase } from '../utils/functions/unCamelCase'
+import Typography from '../utils/typography/Typography'
+import Creatable from 'react-select/creatable'
+import {
+  EnumOption,
+  findValueOption,
+  Control,
+} from './entity/annotations/CustomSelectWidget'
+
+export type QuerySortSelectorProps = {
+  sortConfig: SortConfiguration
+  facetAliases?: Record<string, string>
+}
+
+const QuerySortSelector: React.FunctionComponent<QuerySortSelectorProps> = ({
+  sortConfig,
+  facetAliases,
+}) => {
+  const { defaultColumn, defaultDirection, sortableColumns } = sortConfig
+  const infiniteQueryContext = useInfiniteQueryContext()
+  const { getLastQueryRequest, executeQueryRequest } = infiniteQueryContext
+  const [sortColumn, setSortColumn] = useState<string | undefined>(
+    defaultColumn,
+  )
+  const [sortDirection, setSortDirection] = useState(defaultDirection)
+
+  const enumOptions: EnumOption[] = sortableColumns.map(sortableColumn => {
+    return {
+      value: sortableColumn,
+      label: unCamelCase(sortableColumn, facetAliases)!,
+    }
+  })
+
+  const onChange = (value?: string) => {
+    const lastQueryRequestDeepClone = getLastQueryRequest()
+    let newSortDirection: SortDirection = 'ASC'
+    if (value === sortColumn) {
+      // flip sort direction
+      newSortDirection = sortDirection === 'ASC' ? 'DESC' : 'ASC'
+    }
```

Comment:
More of a design issue--this functionality isn't clear from the UI, in my opinion.

Reply:
Yes, I totally agree.  This is not obvious and my implementation deviates from the design.  The design had the selected item indicating the current sort direction, and another item in the combobox (for the same column) that indicates the opposite sort.  This is not how select boxes work though (the selected item needs to be one of the options in the select combobox).